### PR TITLE
Vc707 nopcie switchable

### PR DESCRIPTION
**Pull request by mcd500**
This pull request is prerequisite for the top pull request of freedom for vc707 with and without PCIe-HMC board as disccusted at #116

### CODE REVIEW

File: xilinx/vc707/constraints/vc707-master.xdc [hunk 1]
```diff
@@ -1,6 +1,3 @@
 #-------------- MCS Generation ----------------------
 set_property CFGBVS GND                               [current_design]
 set_property CONFIG_VOLTAGE 1.8                       [current_design]
-
-set_property EXTRACT_ENABLE YES                       [get_cells dut_/spi_0_1/mac/phy/txd_reg*]
```
Comment (erikdanie) [inline xilinx/vc707/constraints/vc707-master.xdc]: Can you explain to me why you made this change?

**Comment by krabo0om** [comment]
@mcd500 Hey, thank you for your effort. What is the current status of this pull request?

**Comment by erikdanie** [comment]
Hi all, sorry for the long delay, but I will be merging this into master soon, once some other dependencies have cleared up. It should be fine to use in the meantime though, as it looks good functionally

File: xilinx/vc707/constraints/vc707-master.xdc (hunk 1 above)
Comment (mcd500) [reply xilinx/vc707/constraints/vc707-master.xdc]: No technical reason. 
When I made the pull request https://github.com/sifive/freedom/pull/118 I just used where the freedom was pointing at fpga-shells and the commit included the patch for  `Revert "SPI speedup related FPGA changes"` which changing the `set_property EXTRACT_ENABLE YES`.
I am fine being merged without this change.

**Comment by mcd500** [comment]
gentle reminder ping?

File: xilinx/vc707/constraints/vc707-master.xdc (hunk 1 above)
Comment (mcd500) [reply xilinx/vc707/constraints/vc707-master.xdc]: If you want me to regenerate the pull request without the https://github.com/sifive/fpga-shells/commit/a0cc85e01d86bf7a8cd0a312b70da5db12e5d5b9, please let me know.

**Comment by mcd500** [comment]
@erikdanie Thanks for coming back on my pull request. I rebased on master and just tried building it and it gives following error message. The vc707 branch in freedom seems to be far behind. Do you have an idea where to fix the following errors? I would like to make fixes for the pull request.
```
[error] /home/akirat/dev/freedom/src/main/scala/unleashed/DevKitFPGADesign.scala:96:63: not enough arguments for method apply: (baseAddress: BigInt, wrangler: sifive.fpgashells.clocks.ClockAdapterNode, corePLL: sifive.fpgashells.clocks.PLLNode)(implicit p: freechips.rocketchip.config.Parameters)sifive.fpgashells.shell.DDROverlayParams in object DDROverlayParams.
[error] Unspecified value parameter corePLL.
[error]   val ddr = p(DDROverlayKey).headOption.map(_(DDROverlayParams(p(ExtMem).get.master.base, wranglerNode)))
[error]                                                               ^
[error] /home/akirat/dev/freedom/src/main/scala/unleashed/DevKitFPGADesign.scala:97:11: value := is not a member of Nothing
[error]   Expression does not convert to assignment because receiver is not assignable.
[error]   ddr.get := mbus.toDRAMController(Some("xilinxvc707mig"))()
[error]           ^
[error] /home/akirat/dev/freedom/src/main/scala/unleashed/DevKitFPGADesign.scala:105:67: not enough arguments for method apply: (wrangler: sifive.fpgashells.clocks.ClockAdapterNode, bars: Seq[freechips.rocketchip.diplomacy.AddressSet], ecam: BigInt, corePLL: sifive.fpgashells.clocks.PLLNode)(implicit p: freechips.rocketchip.config.Parameters)sifive.fpgashells.shell.PCIeOverlayParams in object PCIeOverlayParams.
[error] Unspecified value parameter corePLL.
[error]   val pcies = p(PCIeOverlayKey).headOption.map(_(PCIeOverlayParams(wranglerNode)))
[error]                                                                   ^
[error] /home/akirat/dev/freedom/src/main/scala/unleashed/DevKitFPGADesign.scala:106:9: Cannot construct a collection of type That with elements of type (A1, Int) based on a collection of type Iterable[Nothing].
[error]   pcies.zipWithIndex.map { case((pcieNode, pcieInt), i) =>

```

**Comment by mcd500** [comment]
I realized that I have encounter this symptom already in the past. https://github.com/sifive/freedom/issues/120
Seriously I would like to help resolving this. I really appreciate for your guide.

**Comment by mcd500** [comment]
@erikdanie Thanks, I was able to resolve the error of ddr and pcies for vc707https://github.com/mcd500/freedom/commit/8c83ba66005e8d6fcc216d3ef6781c48a8ac54c7. Then different error popped up.
```
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:20:42: object verashell is not a member of package sifive.fpgashells.shell.microsemi
[error] import sifive.fpgashells.shell.microsemi.verashell.{VeraShell,HasPCIe,HasDDR3,HasPFEvalKitChipLink}
[error]                                          ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:171:63: not found: type VeraShell
[error] class IOFPGAChip(implicit override val p: Parameters) extends VeraShell
[error]                                                               ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:172:8: not found: type HasPFEvalKitChipLink
[error]   with HasPFEvalKitChipLink {
[error]        ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:192:3: not found: value dut_clock
[error]   dut_clock := hart_clk_125
[error]   ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:192:16: not found: value hart_clk_125
[error]   dut_clock := hart_clk_125

```
I changed the following as a temporary try.
```diff --git a/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala b/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala
index 7a1cfbe..1dca98e 100644
--- a/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala
+++ b/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala
@@ -17,13 +17,15 @@ import sifive.blocks.devices.pinctrl.{BasePin}
 import sifive.blocks.devices.msi._
 import sifive.blocks.devices.chiplink._
 
-import sifive.fpgashells.shell.microsemi.verashell.{VeraShell,HasPCIe,HasDDR3,HasPFEvalKitChipLink}
+import sifive.fpgashells.shell.microsemi._
+//import sifive.fpgashells.shell.microsemi.verashell.{VeraShell,HasPCIe,HasDDR3,HasPFEvalKitChipLink}
 import sifive.fpgashells.devices.microsemi.polarfireevalkitpciex4._
 import sifive.freedom.unleashed.FreedomU500Config
 import sifive.fpgashells.ip.microsemi.CLKINT
 import sifive.fpgashells.ip.microsemi.polarfiredll._
 import sifive.fpgashells.ip.microsemi.polarfireccc._
 
+import sifive.fpgashells.shell.microsemi.polarfireevalkitshell._
 import sifive.fpgashells.devices.microsemi.polarfireddr4._
 import sifive.fpgashells.clocks._
```
Then the message have changed. I am not expert on scala. I am not sure why vera and iofpa comes up for vc707. Would you mind guiding me again? 
```
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:174:8: illegal inheritance;
[error]  self-type sifive.freedom.unleashed.vera.iofpga.IOFPGAChip does not conform to sifive.fpgashells.shell.microsemi.polarfireevalkitshell.HasPFEvalKitChipLink's selftype sifive.fpgashells.shell.microsemi.polarfireevalkitshell.HasPFEvalKitChipLink with sifive.fpgashells.shell.microsemi.polarfireevalkitshell.PolarFireEvalKitShell
[error]   with HasPFEvalKitChipLink {
[error]        ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:194:3: not found: value dut_clock
[error]   dut_clock := hart_clk_125
[error]   ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:194:16: not found: value hart_clk_125
[error]   dut_clock := hart_clk_125
[error]                ^
[error] /var/www/html/git/freedom-mcd500-pull-request/src/main/scala/unleashed/vera/iofpga/FPGAChip.scala:195:3: not found: value dut_ext_reset_n
[error]   dut_ext_reset_n := ereset_n
[error]   ^

```

**Comment by mcd500** [comment]
Anyway the submodule `fpga-shells` do not generate error any more so I created the PR of submodule `fpga-shells` only. https://github.com/sifive/fpga-shells/pull/79

**Comment by mcd500** [comment]
I also pushed the changes reflecting your comment at https://github.com/sifive/fpga-shells/pull/79 to this PR which is on top of sifive:revert-43-fpga-spi_speedup.

**Comment by mcd500** [comment]
Thanks for merging.

**Pull request merged**